bindings/rust/evmc-vm/src/lib.rs

```diff
@@ -243,12 +267,37 @@ impl<'a> ExecutionContext<'a> {
         }
     }
 
-    pub fn call(&mut self, message: &ffi::evmc_message) -> ExecutionResult {
+    pub fn call(&mut self, message: &ExecutionMessage) -> ExecutionResult {
```

Comment:
I am a bit concerned about passing by reference here. It can be extremely weird if the host ends up mutating through a pointer to something that belonged to the `ExecutionMessage`.
Maybe it should just be consumed by value?

Reply:
Well it is passed as a `const` to the host so it should not mutate it, otherwise it violates the EVMC interface.

Reply:
Good point, should be fine then.